lib/build-metadata.js

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env node
+"use strict";
+
+/**
+ * @typedef {Object} metadata
+ * @property {metadataProperty} *
```

Comment:
So our large conversation about documentation came out of this. Besides the scheduled discussion on it, I had the wrong idea about what exactly this is trying to denote and there are a few leaps that need to happen (not a clean trail to find what you are looking for).

1. "package" names used in code are automatically camel cased.
2. `shelfLib` refers to the "package" name, not a module or variable being injected. This sort of refers to the index file but not really because its actually the result of the factory so notation like `package.property` doesn't really ever make sense unless you literally `module.exports.property = X`. Thats ok though because we decided to ALWAYS use `~` anyway.
3. It is assumed this module is somewhere in your package file for tracing purposes.

And after all that, I just removed these typedefs. They don't really make sense because they are already defined in shelfLib.